test/parallel/test-string-decoder.js

```diff
@@ -193,11 +193,8 @@ function unicodeEscape(str) {
 //   [ [ 0, 1 ], [ 1, 3 ] ],
 //   [ [ 0, 1 ], [ 1, 2 ], [ 2, 3 ] ]
 // ]
-function writeSequences(length, start, sequence) {
-  if (start === undefined) {
-    start = 0;
-    sequence = [];
-  } else if (start === length) {
+function writeSequences(length, start = 0, sequence = []) {
+  if (start === length) {
```

Comment:
This isn't strictly the same as before. Given `length = 0, start = undefined, sequence = undefined` it will produce `length = 0, start = 0, sequence = [[]]`.

Reply:
@apapirovski : Thanks for the feedback. I Agree with you but as per method description (writeSequences returns an array of arrays), `[[]]` seemed more correct to me. Kindly advice.

Reply:
Given `length` 0 this function should return `[]` as there would be nothing to write. Anyway, this is all pretty theoretical given that the tests above never pass in `length = 0`.

Reply:
@apapirovski : Understand your thoughts; as you said, its pretty theoretical.Kindly advice on whether we must keep this change or not.. if not, I'll close the PR ! Thanks again for your valuable feedback.

Reply:
@apapirovski : Just to add, I feel in case of `length` 0, the function should return `[[]]` to be consistent with description as it says array of arrays. But I do concede that your points are absolutely right. Indeed quite a tricky thing 😄

Reply:
I won't block it from landing and it has 1 approval so it could land after 48 hours. I'm just offering up my thoughts.

In general, if you were to ask me re: good things to work on, given that you've already contributed quite a bit of code to Node.js (thank you for that, btw!), I would suggest finding gaps in test coverage using https://coverage.nodejs.org/ and creating new tests or expanding existing ones.

Reply:
@apapirovski : Thanks much for your motivation. I think I'm now comfortable with the code base to some extend and plan to take up something better like what you've suggested. I'll definitely get started with what you've referred next 👍 Thanks again for all your support and guidance !

Reply:
@apapirovski : As discussed above, just to confirm...hope this need not be closed and will be landed! Thanks.